src/platform/windows.c

```diff
@@ -1126,7 +1117,7 @@ static bool windows_setup_file_association(
 	}
 	// [hRootKey\OpenRCT2.ext\DefaultIcon]
 	wchar_t szIconW[MAX_PATH];
-	wsprintfW(szIconW, L"\"%s\",%d", dllPathW, iconIndex);
+	StringCchPrintfW(szIconW, MAX_PATH, L"\"%s\",%d", dllPathW, iconIndex);
```

Comment:
> even if the string is longer the the buffer size

I'm not sure that's particularly ok 😜. Either way, the different handling is that, when the resultant string is longer than the buffer, the buffer is set to an empty string (and maybe an error number is returned).